pkg/kubectl/apis/manifest/v1alpha1/types.go

```diff
@@ -0,0 +1,123 @@
+
+	// Labels to add to all objects and selectors.
+	// These labels would also be used to form the selector for apply --prune
+	// Named differently than “labels” to avoid confusion with metadata for this object
+	ObjectLabels map[string]string `json:"objectLabels,omitempty" yaml:"objectLabels,omitempty"`
+
+	// Annotations to add to all objects.
+	ObjectAnnotations map[string]string `json:"objectAnnotations,omitempty" yaml:"objectAnnotations,omitempty"`
+
+	// Partial name that will prefix the name of the base resources.
+	NamePrefix string `json:"namePrefix,omitempty" yaml:"namePrefix,omitempty"`
+
+	// Description of the application package.
+	Description string `json:"description,omitempty" yaml:"description,omitempty"`
+
+	// Search keywords for this application package.
+	Keywords []string `json:"keywords,omitempty" yaml:"keywords,omitempty"`
+
+	// Version of the application package.
+	AppVersion string `json:"appVersion,omitempty" yaml:"appVersion,omitempty"`
+
+	// Homepage of the application package.
+	Home string `json:"home,omitempty" yaml:"home,omitempty"`
+
+	// Sources specify the URLs, e.g. https://github.com/foo/bar.git, file://host/path, etc.
+	// hosting the resource files specified in Base and Overlays.
+	Sources []string `json:"sources,omitempty" yaml:"sources,omitempty"`
+
+	// An pointer to the icon.
+	Icon string `json:"icon,omitempty" yaml:"icon,omitempty"`
+
+	// People that maintain this package.
+	Maintainers []Maintainer `json:"maintainers,omitempty" yaml:"maintainers,omitempty"`
+
+	// A Base entry specifies the file path, relative to Source, of an API resource definition.
+	Bases []string `json:"bases,omitempty" yaml:"bases,omitempty"`
+
+	// An Overlay entry specifies the file path, relative to Source, of a file
+	// containing a Strategic Merge Patch overlay in API resource form.
+	Overlays []string `json:"overlays,omitempty" yaml:"overlays,omitempty"`
+
+	// List of configmap overlays.
+	Configmaps []ConfigMap `json:"configmaps,omitempty" yaml:"configmaps,omitempty"`
+
+	// List of secret overlays.
+	Secrets []Secret `json:"secrets,omitempty" yaml:"secrets,omitempty"`
+
+	// Whether PersistentVolumeClaims should be deleted with the other resources.
+	OwnPersistentVolumeClaims bool `json:"ownPersistentVolumeClaims,omitempty" yaml:"ownPersistentVolumeClaims,omitempty"`
+
+	// Whether recursive look for Kube-manifest.yaml, similar to `kubectl --recursive` behavior.
+	Recursive bool `json:"recursive,omitempty" yaml:"recursive,omitempty"`
+
+	// Whether prune resources not defined in Kube-manifest.yaml, similar to `kubectl apply --prune` behavior.
+	Prune bool `json:"prune,omitempty" yaml:"prune,omitempty"`
+}
+
+// ConfigMap contains the metadata of using a configmap.
+type ConfigMap struct {
+	// The type of the configmap. e.g. `env` and `file`.
+	Type string `json:"type,omitempty" yaml:"type,omitempty"`
+
+	// Partial name that will prefix the configmap overlays.
```

Comment:
Look at `kubectl create configmap --append-hash` to understand what parameters are needed and what they mean